verification_token/models.py

```diff
@@ -18,20 +19,26 @@ class VerificationTokenManager(models.Manager):
     def deactivate(self, obj, slug=None, key=None):
         self.filter_active_tokens(obj, slug, key).update(is_active=False)
 
-    def deactivate_and_create(self, obj, slug=None, deactivate_old_tokens=True, expiration_in_minutes=None,
+    def deactivate_and_create(self, obj, slug=None, extra_data=None, deactivate_old_tokens=True, expiration_in_minutes=None,
                               key_generator_kwargs=None):
         expiration_in_minutes = settings.DEFAULT_EXPIRATION if expiration_in_minutes is None else expiration_in_minutes
 
         key_generator_kwargs = {} if key_generator_kwargs is None else key_generator_kwargs
         if deactivate_old_tokens:
             self.deactivate(obj, slug)
-        return self.create(
+
+        token = self.create(
             content_type=ContentType.objects.get_for_model(obj.__class__),
             object_id=obj.pk,
             slug=slug,
             expiration_in_minutes=expiration_in_minutes,
-            key=self.model.generate_key(**key_generator_kwargs)
+            key=self.model.generate_key(**key_generator_kwargs),
+            extra_data=pickle.dumps(extra_data),
         )
+        if extra_data:
+            token.set_extra_data(extra_data)
```

Comment:
buď zruš nastavení extra dat výše nebo tady. Také by se mi líbílo, kdybys to měl v jedné metodě. Takže raději připrav objekt (nepoužívej self.create, ale self.model(xxx) a následně token.set_extra_data() a nakonec token.save()

Reply:
Jo to výše je zapomenuté, díky. Tobě se teda nelíbí dvojásobné uložení?